### PR TITLE
feat: tabs-component

### PR DESCRIPTION
**Pull request by georgeciubotaru**

### CODE REVIEW

**Comment by zolotokrylin** [comment]
Related issue: https://github.com/amplify-labs/factor-app/pull/47

**Comment by zolotokrylin** [comment]
@tsejerome @georgeciubotaru @markholdex please remember to log the reference (like this https://github.com/zolotokrylin/svelboost/pull/2#issuecomment-899979248) to the relevant epic/issue when it is not possible to connect them via UI in GitHub because they are from diff repos. Thanks!

**Comment by zolotokrylin** [comment]
@markholdex @georgeciubotaru how can we use the Storybooks. I don't see storybooks integration in this PR.

File: src/components/Tabs/Tabs.svelte [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    import TabsHead from "./TabsHead.svelte";
+    import TabItem from "./TabItem.svelte";
+
+    type Item = {
+        label: string;
+        href: string;
+        onClick?: () => void;
+    };
+
+    export let defaultActive: string = "0";
```
Comment (zolotokrylin) [inline src/components/Tabs/Tabs.svelte]: @tsejerome why string, not int? Is it the index of the active tab?

File: src/components/Tabs/Tabs.svelte [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    export let skipBorder: boolean = false;
+    export let items: Item[];
+
+    let className: string = "";
+    export { className as class };
+
+    let tab = writable<string>(defaultActive);
+    $: $tab = defaultActive;
```
Comment (zolotokrylin) [inline src/components/Tabs/Tabs.svelte]: @tsejerome why does it needs to be reactive? ($:)

File: src/components/Tabs/Tabs.svelte [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    let handleTabClick = (index: number, clbk: any) => {
+        tab.set(index.toString());
+        if (clbk) clbk();
+    };
+</script>
+
+<div class="tabs {className}">
+    <TabsHead skipBorder={skipBorder}>
```
Comment (zolotokrylin) [inline src/components/Tabs/Tabs.svelte]: @tsejerome why do we need TabsHead? looks like overhead to me.
If there is head, then there is a body? But I can see only Head, which looks like a decorative wrapper around. Or does it have a functional purpose as well?

**Review by zolotokrylin — CHANGES_REQUESTED** [review]
Guys, I need a consultation with you. I think there are overheads here. I stopped putting inline comments as I have too many.

Also, why do we need to export so many components to the Tabs component consumer? 
Can consumers just import `Tabs` and supply an array of tabs properties? The end consumer component is Tabs, not tabs Header, Item, etc.

File: src/components/Tabs/Tabs.svelte (hunk 1 above)
Comment (tsejerome) [reply src/components/Tabs/Tabs.svelte]: Yes it is the index of the active tab. I also agree that it is weird to be a string too.

I think that it shd be
1) renamed as `defaultActiveTabIndex`
2) type of `number` 

I tried to do the development with minimal changes so I didn't change the original code copied from holdex
If it is okay to have more modifications then I will rename and refactor this part

**Comment by zolotokrylin** [comment]
Log: Discussed with Jerome during the call. Jerome will make the adjustments to the tabs.

**Comment by tsejerome** [comment]
@zolotokrylin Just updated as per our previous discussion, please review thanks!

Discovered a current issue, on `src/components/Tabs/Tabs.svelte `, if I switch `defaultActiveItemIndex` to non-reactive, it seem that it won't load the default tab as underscored successfully
image

File: src/components/Tabs/Tabs.svelte [hunk 4]
```diff
@@ -8,18 +8,17 @@
...
+    export let defaultActiveItemIndex: number = 0;
     export let items: Item[];
 
     let className: string = "";
     export { className as class };
 
-    let tab = writable<string>(defaultActive);
-    $: $tab = defaultActive;
+    let tab = writable<number>(defaultActiveItemIndex);
+    $: tab.update(a=> defaultActiveItemIndex);
```
Comment (zolotokrylin) [inline src/components/Tabs/Tabs.svelte]: discussed in chat.

**Review by zolotokrylin — CHANGES_REQUESTED** [review]
log: Discussed issues with Jerome in chat. He will apply changes.

@tsejerome try to wrap up this PR by today. Tomorrow is already Friday. If needed let's have a quick call to speed it up.

File: src/components/Tabs/Tabs.svelte (hunk 4 above)
Comment (zolotokrylin) [reply src/components/Tabs/Tabs.svelte]: confusion between `$:` and `$var` was resolved.

File: src/components/Tabs/TabHead.svelte [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<script lang="ts">
+    export let maxWidth: number = undefined;
+
+    $: headStyle = maxWidth ? `width: ${maxWidth}px;` : "";
```
Comment (zolotokrylin) [inline src/components/Tabs/TabHead.svelte]: why reactive?

**Review by zolotokrylin — APPROVED** [review]

**Pull request merged**